Cargo.toml

```diff
@@ -10,14 +10,21 @@ description = "General purpose library for using and interoperating with Bitcoin
 keywords = [ "crypto", "bitcoin" ]
 readme = "README.md"
 
+[[example]]
+name = "bip32"
+required-features = ["secp256k1"]
+
+[[example]]
+name = "handshake"
+required-features = ["secp256k1"]
 
 [features]
 default = [ "secp-recovery" ]
 base64 = [ "base64-compat" ]
 fuzztarget = ["secp256k1/fuzztarget", "bitcoin_hashes/fuzztarget"]
 unstable = []
 rand = ["secp256k1/rand-std"]
-use-serde = ["serde", "bitcoin_hashes/serde", "secp256k1/serde"]
```

Comment:
I think that with the removal of this feature, users will now have to explicitly depend on `secp256k1` if they want to enable the `serde` feature. That in turn means that the versions need to be kept in sync which makes usage of services like `dependabot` really cumbersome because they can no longer bump the dependencies correctly.

Given that we already have a set of `secp-` features, I would propose to add a `secp-serde` feature to restore this functionality.

Reply:
I'm curious whether `use-serde` and `secp256k1` without `secp-serde` work correctly. It's really unfortunate that we have such a largo feature matrix :(.

Reply:
It at least compiles and I can't imagine what the code would have to do to behave differently. I didn't touch behavior (outside skipping tests), just conditionally disabled items.

`secp-` sounds like a good idea.

Reply:
@apoelstra A potential solution for this was actually just merged 25 days ago (it's still an unstable cargo feature obviously)
https://github.com/rust-lang/cargo/pull/8818

Reply:
@elichai that's nice! Shame it's only in the new Cargo but at least there's hope that one day this will be solved more cleanly. :)

I just added `secp-serde`